### PR TITLE
Remove *Search gRPC methods and cleanup search

### PR DESCRIPTION
**Pull request by jotare**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nidx/nidx_protos/src/nidx.proto [hunk 1]
```diff
@@ -33,7 +33,6 @@ service NidxSearcher {
   rpc DocumentSearch(nodereader.DocumentSearchRequest) returns (nodereader.DocumentSearchResponse) {}
   rpc ParagraphSearch(nodereader.ParagraphSearchRequest) returns (nodereader.ParagraphSearchResponse) {}
   rpc VectorSearch(nodereader.VectorSearchRequest) returns (nodereader.VectorSearchResponse) {}
-  rpc RelationSearch(nodereader.RelationSearchRequest) returns (nodereader.RelationSearchResponse) {}
```
Comment (javitonino) [inline nidx/nidx_protos/src/nidx.proto:36]: Maybe we can remove DocumentSearch, ParagraphSearch, etc. as well?

File: nucliadb/src/nucliadb/search/search/shards.py [hunk 2]
```diff
@@ -61,11 +61,41 @@ async def get_shard(node: AbstractIndexNode, shard_id: str) -> Shard:
 async def query_paragraph_shard(
     node: AbstractIndexNode, shard: str, query: ParagraphSearchRequest
 ) -> ParagraphSearchResponse:
-    req = ParagraphSearchRequest()
-    req.CopyFrom(query)
-    req.id = shard
+    req = SearchRequest()
```
Comment (javitonino) [inline nucliadb/src/nucliadb/search/search/shards.py:64]: This is called in a single place `resource_search`. Maybe it makes more sense to put this logic (or a subset of it) in `resource_search` and remove this method altogether?

**Review by javitonino — APPROVED** [review]
Nice. Left a comment for a little further cleanup if you feel like doing it now (we can always do a second pass at some point)

File: nidx/nidx_protos/src/nidx.proto (hunk 1 above)
Comment (jotare) [reply nidx/nidx_protos/src/nidx.proto:36]: Yes, I was working on it :)

File: nucliadb/src/nucliadb/search/search/shards.py (hunk 2 above)
Comment (jotare) [reply nucliadb/src/nucliadb/search/search/shards.py:64]: Yes, still working on it. The first step was to remove the calls in the innermost point and go bottom to top

File: nidx/nidx_protos/src/nidx.proto (hunk 1 above)
Comment (jotare) [reply nidx/nidx_protos/src/nidx.proto:36]: Done

File: nucliadb/src/nucliadb/search/search/shards.py (hunk 2 above)
Comment (jotare) [reply nucliadb/src/nucliadb/search/search/shards.py]: Done

**Review by javitonino — APPROVED** [review]

**Pull request merged**